### DISCUSSION TITLE
Generate SSL certificate inside Coolify app

### DISCUSSION THREAD

**Discussion by alexbash7** [coollabsio/coolify#1225] [Feature Requests]
Hi, 

I really like Coolify v4 for it modern design and great UX. I'm currently using Caprover in my projects and would love to switch to Coolify, but I can't find one important feature - ability to generate free SSL certificate for the app right in the Collify app

**Comment by andrasbacsai** [comment]
> Hi, 
> 
> 
> 
> I really like Coolify v4 for it modern design and great UX. I'm currently using Caprover in my projects and would love to switch to Coolify, but I can't find one important feature - ability to generate free SSL certificate for the app right in the Collify app

Hey. If you use Traefik proxy (the default) SSL certificates are automatically generated for you if you set https to your application.

This is what are you looking for?

**Reply by JJDSNT** [reply]
hi, i tried to setup a nextjs using coolify on oracle cloud.
i could access coolify at port 8000, but i dont know how to access the nextjs app in the port 3000.
in the log it says it is serving at localhost:3000
what would be the settings that i need to have it accessible from a public url/ip?
(sorry for the newbie question but cant found much info about the v4 over the web)

Best regards,

**Reply by andrasbacsai** [reply]
You need to enable ports 80 and 443 in Oracle Cloud’s dashboard to be able to access your app with domain through Coolify’s proxy.

If you are mapping the port to the host system, then you need to allowe that port in th same Oracle dashboard.

Oracle by default does not allow inbound requests. You need to enable them manually.

**Comment by alexbash7** [comment]
Yes, it works! Thanks

**Reply by itxtoledo** [reply]
EDIT: after 4-5 minutes the SSL appears

How did you manage to make this work? I changed the domain to start with https, restarted the application and in the browser my domain still doesn't have SSL

**Comment by zyjswiadomielife** [comment]
I also don't have automatic ssl on coolify. I try on few vps and still nothing. My domain pointed to vps and still nothing. No ssl on instance domain and no ssl on application after deploy(I wait even one day and still nothing). System is ubuntu, I try on Ubuntu 20 and 22 version.

**Reply by ohyajk** [reply]
hello @zyjswiadomielife after you point your domain to IP. add htttps in the domain section in your coolify project and restart the project.
![https domain coolify](https://github.com/user-attachments/assets/abe6a93f-0d6c-4da7-a9ba-17c6a8a1e063)

**Reply by MentalGear** [reply]
I had to change the input, then enter the domain again into the field for SSL to work. Seems like a small bug, but overall there should be an own section for SSL because it's not at all clear how it works to set it up, even if automatic.
